src/core/sys/posix/sys/time.d

```diff
@@ -28,7 +28,55 @@ else version (WatchOS)
     version = Darwin;
 
 version (Posix):
-extern (C) nothrow @nogc:
+nothrow @nogc:
+
+
+/* macros provided to operate on timeval structures
+ * they are extern(D) because they are not really C symbols, just macros
+ */
+
+pure @safe void timeradd(const timeval* a, const timeval* b,
+        timeval* result)
+{
+    result.tv_sec = a.tv_sec + b.tv_sec;
+    result.tv_usec = a.tv_usec + b.tv_usec;
```

Comment:
> otherwise, the result doesn't make sense.

Well, it makes sense, just a weird sense, just the same as having usecs > 1_000_000. This is based on a C macro, I know is not as purely correct as one might think, but we are wrapping C here, so I'm not sure if it's a good idea to depart from it. If we want better timer manipulation functions, they could be written in D, but this is just an interface to C functions.

But looking into this, I just realized this is not really POSIX:
```
CONFORMING TO

       Not in POSIX.1.  Present on most BSD derivatives.
```

http://man7.org/linux/man-pages/man3/timeradd.3.html

OK, I have to rethink this, and probably move it to the linux/bsd specific bindings...